service/src/main/java/bio/terra/axonserver/app/controller/GetFileController.java

```diff
@@ -64,16 +66,34 @@ public ResponseEntity<byte[]> getFile(
    * @return - A ResponseEntity containing the file
    */
   @Override
-  public ResponseEntity<byte[]> getFileInBucket(
+  public ResponseEntity<Resource> getFileInBucket(
       UUID workspaceId, UUID resourceId, String objectPath, @Nullable String convertTo) {
+    return buildResponse(workspaceId, resourceId, objectPath, convertTo);
+  }
 
+  private ResponseEntity<Resource> buildResponse(
+      UUID workspaceId, UUID resourceId, @Nullable String objectPath, @Nullable String convertTo) {
     BearerToken token = getToken();
 
     HttpRange byteRange = getByteRange();
 
-    byte[] resourceObj =
+    File resourceObj =
         fileService.getFile(token, workspaceId, resourceId, objectPath, convertTo, byteRange);
-    return new ResponseEntity<>(resourceObj, HttpStatus.OK);
+    String contentType =
+        convertTo == null
+            ? URLConnection.guessContentTypeFromName(resourceObj.getPath())
+            : URLConnection.guessContentTypeFromName((resourceObj.getName() + "." + convertTo));
```

Comment:
Is there an injection risk here? The `convertTo` is taken straight from the request and could be any string, right?
Is there validation somewhere?

Reply:
Thanks for the quick turnaround! Yes, but the previous line that calls fileService will throw an InvalidConvertToFormat 400 error if the convertTo value isn't supported, e.g if its anything other than "html".